### DISCUSSION TITLE
Lichess speed optimization

### DISCUSSION THREAD

**Discussion by cyberluke** [EdNekebno/DGTCentaurMods#86] [General]
Hi, I would say I have expert knowledge with networking, integrations, APIs, socket, websocket, messaging queues, polling.

I did notice Lichess is a bit slow. Did you do any research what could be a bottleneck, so I could help with it?

We should be able to get same experience as mobile apps. I did not study it yet, but there might be incorrect use of Lichess API or wrong timing of request wgich would result in http requests rate limiting from Lichess side (like timeouts and overusing polling). But that is just my blind guess from a train (Im actually on train now haha)

**Comment by cyberluke** [comment]
I briefly look in the code and could not find any specific Lichess API use. But I guess it is using http rest api, not websocket. I should implement websocket api.

**Reply by wormstein** [reply]
The key is used in game/lichess.py
